pkg/trace/agent/fuzz_test.go

```diff
@@ -0,0 +1,121 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+//go:build go1.18
+
+package agent
+
+import (
+	"testing"
+
+	"github.com/DataDog/datadog-agent/pkg/trace/pb"
+	"github.com/DataDog/datadog-agent/pkg/trace/testutil"
+)
+
+func FuzzProcessStats(f *testing.F) {
+	agent, cancel := agentWithDefaults()
+	defer cancel()
+	encode := func(pbStats pb.ClientStatsPayload) ([]byte, error) {
+		return pbStats.Marshal()
+	}
+	decode := func(stats []byte) (pb.ClientStatsPayload, error) {
+		var payload pb.ClientStatsPayload
+		err := payload.Unmarshal(stats)
+		return payload, err
+	}
+	pbStats := testutil.StatsPayloadSample()
+	stats, err := encode(pbStats)
+	if err != nil {
+		f.Fatal("Couldn't generate seed corpus:", err)
```

Comment:
nit: I believe this won't put a space between the colon and the error. I tend to see these more like `f.Fatalf("Couldn't generate seed corpus: %v", err)` instead as a best practice for formatting.

Reply:
Actually a space will be added because `f.Fatal` calls `fmt.Sprintln` to format the arguments

```go
// Fatal is equivalent to Log followed by FailNow.
func (c *common) Fatal(args ...any) {
	c.checkFuzzFn("Fatal")
	c.log(fmt.Sprintln(args...))
	c.FailNow()
}
```

But I'll switch to `f.Fatalf` since it's a best practice for formatting, thanks!

Reply:
Ah I even looked at the code to double check, and still didn't see the `Sprintln`. My bad. 🤦 
Thanks!